doc/langref.html.in

```diff
@@ -5088,6 +5088,46 @@ fn deferUnwindExample() void {
 
 test "defer unwinding" {
     deferUnwindExample();
+}
+      {#code_end#}
+      {#code_begin|test|test_mutable_vars_defer#}
+const std = @import("std");
+
+const expect = std.testing.expect;
+
+// It's important to understand the semantics of `defer` when using mutable
+// variables. The `defer` statement captures the value of the variable at the
+// time the `defer` statement is executed, which is at the end of the current
+// scope. This means that if the variable is mutated after the `defer` statement
+// is defined but before the end of the scope, the `defer` statement will
+// execute with the mutated value.
+test "defer with mutable variables" {
+    var x: u8 = 42;
+    defer expect(x == 69) catch unreachable;
```

Comment:
Please use `std.debug.assert` here.

Reply:
@notcancername Why? You mind elaborating?

Reply:
@notcancername Actually, I see why—it does appear to be the better option in this case. I'll update it. Thank you.

Reply:
neither are good as both will crash the test runner if failed

Reply:
Sorry, what else do you suggest for use with `defer`?

Reply:
oh this is in the langref. that's still tested but I suppose my point wasn't great per se because this will always pass